workspaces/rbac/plugins/rbac-backend/src/admin-permissions/admin-creation.ts

```diff
@@ -162,13 +162,46 @@ const addAdminPermissions = async (
   }
 };
 
+const removeOldCreateAdminPermissions = async (
+  enf: EnforcerDelegate,
+  auditor: AuditorService,
+) => {
+  const policyEntityCreate = [
+    'role:default/rbac_admin',
+    'policy-entity',
+    'create',
+    'allow',
+  ];
+  if (await enf.hasPolicy(...policyEntityCreate)) {
+    const auditorEvent = await auditor.createEvent({
+      eventId: PermissionEvents.POLICY_WRITE,
+      severityLevel: 'medium',
+      meta: { actionType: ActionType.DELETE, source: 'configuration' },
+    });
+
+    try {
+      await enf.removePolicy(policyEntityCreate);
+      await auditorEvent.success({
+        meta: { policy: policyEntityCreate },
+      });
+    } catch (error) {
+      await auditorEvent.fail({
+        error,
+        meta: { policy: policyEntityCreate },
+      });
+    }
+  }
+};
+
 export const setAdminPermissions = async (
   enf: EnforcerDelegate,
   auditor: AuditorService,
 ) => {
+  // TODO: Temporary workaround to prevent breakages after the removal of the resource type `policy-entity` from the permission `policy.entity.create`
```

Comment:
What do you think how many major versions we are going to wait, while we will remove this code ?

Reply:
I was thinking possibly an RHDH release to remove all references of `policy-entity, create`, that should give everyone plenty of time to make updates. We could also do this is steps as well, where we remove the admin creation and rest api todos first and then follow up with removal of the workaround in permission policy. I can create an issue for discussion.

Reply:
File an issue, please.